cql3/functions/functions.cc

```diff
@@ -413,13 +413,15 @@ functions::get(data_dictionary::database db,
     });
 
     if (similarity_fun_it != SIMILARITY_FUNCTIONS.end()) {
+        auto arg_types = vector_similarity_fct::provide_arg_types(name, provided_args, db);
+
         shared_ptr<vector_similarity_fct> fun;
         if (*similarity_fun_it == SIMILARITY_COSINE_FUNCTION_NAME) {
-            fun = ::make_shared<similarity_cosine_fct>();
+            fun = ::make_shared<similarity_cosine_fct>(arg_types);
         } else if (*similarity_fun_it == SIMILARITY_EUCLIDEAN_FUNCTION_NAME) {
-            fun = ::make_shared<similarity_euclidean_fct>();
+            fun = ::make_shared<similarity_euclidean_fct>(arg_types);
         } else if (*similarity_fun_it == SIMILARITY_DOT_PRODUCT_FUNCTION_NAME) {
-            fun = ::make_shared<similarity_dot_product_fct>();
+            fun = ::make_shared<similarity_dot_product_fct>(arg_types);
```

Comment:
This should have been part of the previous patch. It's just confusing to review something broken that is later fixed.